### PR TITLE
Lvrubym 223: Add our day report model, rspec for model, routes, conroller.

### PR DESCRIPTION
**Pull request by yaremakleputs**
## JIRA

* [Main JIRA ticket](http://ssu-jira.softserveinc.com/browse/LVRUBYM-223)
* [JIRA subtask](http://ssu-jira.softserveinc.com/browse/LVRUBYM-223)

## Code reviewers

- [ ] @skrypnyk88 
- [ ] @andriyarterchuk 
- [ ] @vikabelous 

### Second Level Review

- [ ] @omaik

## Summary of issue

Need to ad back-end of our day report

## Summary of change

- added back-end without controller spec to our day report

## Testing approach

We do not need any tests here

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=90%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: app/controllers/v1/our_days_controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+module V1
+  class OurDaysController < ApplicationController
+    def index
+      @our_days = OurDay.all
```
Comment (skrypnyk88) [inline app/controllers/v1/our_days_controller.rb]: Please limit this for group scope.

File: app/controllers/v1/our_days_controller.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      @our_day = OurDay.new(our_day_params)
+      if @our_day.save
+        render :show
+      else
+        head :bad_request
+      end
+    end
+
+    def show
+      @our_day = OurDay.find(params[:id])
```
Comment (skrypnyk88) [inline app/controllers/v1/our_days_controller.rb]: find(params[:id]   - 1) use find_by
                               2) define in filter

File: app/views/v1/our_days/index.json.jbuilder [hunk 3]
```diff
@@ -0,0 +1,3 @@
+json.array! @our_days, :id,
+                       :description,
+                       :day
```
Comment (skrypnyk88) [inline app/views/v1/our_days/index.json.jbuilder]: Add a blank line. Please set to automatically check in sablaym and check before with robocop.

File: app/views/v1/our_days/show.json.jbuilder [hunk 4]
```diff
@@ -0,0 +1,3 @@
+json.(@our_day, :id,
+                :description,
+                :day)
```
Comment (skrypnyk88) [inline app/views/v1/our_days/show.json.jbuilder]: Same here.

File: config/routes.rb [hunk 5]
```diff
@@ -1,5 +1,6 @@
 Rails.application.routes.draw do
   namespace :v1 do
+    resources :our_days
```
Comment (skrypnyk88) [inline config/routes.rb]: Define from group

Comment (skrypnyk88) [inline config/routes.rb]: Add your code below others if this is possible.

File: db/migrate/20170401170217_create_our_day.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+class CreateOurDay < ActiveRecord::Migration[5.0]
+  def change
+    create_table :our_days do |t|
+      t.text :description
+      t.date :day
```
Comment (skrypnyk88) [inline db/migrate/20170401170217_create_our_day.rb]: Where associations for student and group?

File: db/schema.rb [hunk 7]
```diff
@@ -1,4 +1,17 @@
+# This file is auto-generated from the current state of the database. Instead
+# of editing this file, please use the migrations feature of Active Record to
+# incrementally modify your database, and then regenerate this schema definition.
```
Comment (skrypnyk88) [inline db/schema.rb]: Pleace delete all of this commits

File: db/schema.rb [hunk 8]
```diff
@@ -8,6 +21,13 @@
     t.datetime "updated_at", null: false
   end
 
+  create_table "our_days", force: :cascade do |t|
+    t.text     "description"
+    t.date     "day"
+    t.datetime "created_at",  null: false
```
Comment (skrypnyk88) [inline db/schema.rb]: Where is indexer?

**Comment by skrypnyk88** [comment]
You need rebase!!

File: db/schema.rb [hunk 9]
```diff
@@ -1,4 +1,7 @@
-ActiveRecord::Schema.define(version: 20170403084317) do
+# It's strongly recommended that you check this file into your version control system.
```
Comment (skrypnyk88) [inline db/schema.rb]: Pleace delte this comment line

File: spec/factories/our_days.rb [hunk 10]
```diff
@@ -0,0 +1,8 @@
+require 'faker'
+
+FactoryGirl.define do
+  factory :our_day do
+    description Faker::Lorem.paragraph
```
Comment (skrypnyk88) [inline spec/factories/our_days.rb]: where is group?

File: app/controllers/v1/our_days_controller.rb [hunk 11]
```diff
@@ -0,0 +1,50 @@
+module V1
+  class OurDaysController < ApplicationController
```
Comment (skrypnyk88) [inline app/controllers/v1/our_days_controller.rb]: Where is tests for controller?

File: app/controllers/v1/our_days_controller.rb [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+    before_action :find_groups, only: [:index, :create]
+    before_action :join_our_day, only: [:show, :update, :destroy]
```
Comment (skrypnyk88) [inline app/controllers/v1/our_days_controller.rb]: Maybe better name of this function will be find_our_day? Like in top.

File: app/controllers/v1/our_days_controller.rb [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+    def destroy
+      @our_day.destroy
+      head :ok
+    end
+
+    private
+
+    def join_our_day
+      @our_day = OurDay.find_by(id: params[:id])
```
Comment (skrypnyk88) [inline app/controllers/v1/our_days_controller.rb]: Define in scope of group

File: config/routes.rb [hunk 14]
```diff
@@ -6,6 +6,7 @@
   namespace :v1 do
     resources :groups do
       resources :students, shallow: true
+      resources :our_days, shallow: true
```
Comment (skrypnyk88) [inline config/routes.rb]: I think we don't need this (shallow: true)

File: db/migrate/20170401170217_create_our_day.rb [hunk 15]
```diff
@@ -0,0 +1,10 @@
+class CreateOurDay < ActiveRecord::Migration[5.0]
```
Comment (skrypnyk88) [inline db/migrate/20170401170217_create_our_day.rb]: Is this conventional name for migration?

File: db/migrate/20170410155717_add_index_to_our_day.rb [hunk 16]
```diff
@@ -0,0 +1,5 @@
+class AddIndexToOurDay < ActiveRecord::Migration[5.0]
```
Comment (skrypnyk88) [inline db/migrate/20170410155717_add_index_to_our_day.rb]: The same.

File: db/schema.rb [hunk 17]
```diff
@@ -1,5 +1,7 @@
-ActiveRecord::Schema.define(version: 20170403084317) do
-  # These are extensions that must be enabled in order to support this database
+
+
```
Comment (andriyarterchuk) [inline db/schema.rb]: you should delete this empty lines

File: app/controllers/v1/our_days_controller.rb [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+      @our_days = @group.our_days
+    end
+
+    def create
+      @our_day = @group.our_days.build(our_day_params)
+      if @our_day.save
+        render :show
+      else
+        head :bad_request
+      end
```
Comment (andriyarterchuk) [inline app/controllers/v1/our_days_controller.rb]: you repeated this block in create and update action.
you should create a private method

**Pull request closed without merging**